packages/react-instantsearch-core/src/connectors/connectDynamicWidgets.ts

```diff
@@ -0,0 +1,25 @@
+import createConnector from '../core/createConnector';
+import { getResults } from '../core/indexUtils';
+
+export default createConnector({
+  displayName: 'AlgoliaDynamicWidgets',
+
+  getProvidedProps(props, _searchState, searchResults) {
+    const results = getResults(searchResults, {
+      ais: props.contextValue,
+      multiIndexContext: props.indexContextValue,
+    });
+
+    if (!results) {
+      return { attributesToRender: [] };
+    }
+
+    // retrieve the facet order out of the results:
+    // results.facetOrder.map(facet => facet.attribute)
+    const facetOrder = [];
+
+    return {
+      attributesToRender: props.transformItems(facetOrder, { results }),
```

Comment:
`transformItems` won't be required once the engine implements it. So if we make it optional right now (defaulted to the identity function), we won't have to change our code and APIs to support it, it'll be transparent for both us and users.

Users won't have to upgrade the library for their code and index configuration to work.

So I think we should make `transformItems` optional right now.

Reply:
If it's optional now, we can't change the default, since that's a breaking change, however _not having a default_ changing to having a default later isn't a breaking change, right?

Reply:
The default will always be to return the facet ordering coming from the engine, right? (meaning the breaking change will happen in the index configuration, which is outside of InstantSearch's responsibility.)

I see that we hardcode `const facetOrder = [];`, isn't the engine response specs final yet? (Can't we retrieve e.g., `response.facetOrder || []`?)